tests/ui/methods/clone-missing.rs

```diff
@@ -0,0 +1,19 @@
+// This test checks that calling `.clone()` on a type that does not implement the `Clone` trait
+// results in a compilation error. The `Foo` struct does not derive or implement `Clone`,
+// so attempting to clone it should fail.
+
+struct Foo {
+  i: isize,
+}
+
+fn foo(i:isize) -> Foo {
+    Foo {
+        i: i
+    }
+}
+
+fn main() {
+    let x = foo(10);
+    let _y = x.clone();
+    //~^ ERROR no method named `clone` found
+}
```

Comment:
discussion:
I think let y = x expresses the intention of the test better than let _y = x.clone().
However, if we simply write it like this:
```rust
fn main() {
    let x = foo(10);
    let _y = x;
}
```
we won't get the expected error.

What do you think about writing the test like this instead?

```rust
fn main() {
    let x = foo(10);
    let _y = x;
    let _z = x;
    //~^ ERROR use of moved value: `x`
}
```

Reply:
Got it! Thanks for the context on the original intent! I’ll keep the current test